virttest/libvirt_vm.py

```diff
@@ -1197,12 +1197,14 @@ def add_cputune(vcpu_cputune=""):
                 virt_install_cmd += add_pcidevice(help_text, pci_id)
 
         for image_name in params.objects("images"):
+            basename = False
             image_params = params.object_params(image_name)
 
             base_dir = image_params.get("images_base_dir",
                                         data_dir.get_data_dir())
-
-            basename = params.get("storage_type") == "nfs"
+            if params.get("storage_type") == "nfs":
+                basename = True
+                base_dir = params["nfs_mount_dir"]
```

Comment:
How about also give default value for base_dir like other params?

Reply:
when user choose "nfs" as storage, the directory path is expected to be configured by user. so it is better to error out early and appropriately, instead of assuming with default values.